google/cloud/spanner/keys.h

```diff
@@ -222,6 +222,12 @@ class KeySet {
    */
   KeySet() = default;
 
+  /**
+   * Constructs a `KeySet` from a `spanner::v1::KeySet` protobuf.
+   */
+  explicit KeySet(google::spanner::v1::KeySet key_set)
```

Comment:
As we discussed offline, we probably want to remove this public constructor and instead have an internal::FromProto to match the existing ToProto function.

... Or you could remove this constructor from this PR since it's not related to the equality operations being added here. 

Your call.

Reply:
Added FromProto and made constructor private.